lib/measured/conversion_builder.rb

```diff
@@ -0,0 +1,35 @@
+class Measured::ConversionBuilder
+  def initialize
+    @base_unit = nil
+    @units = []
+    @case_sensitive = false
+  end
+
+  def base(unit_name, aliases: [])
+    raise Measured::UnitError, "Setting #{unit_name} as base unit, but already defined as #{@base_unit}." if @base_unit
+    @base_unit = Measured::Unit.new(unit_name, aliases: aliases)
+  end
+
+  def unit(unit_name, aliases: [], value:)
+    @units << Measured::Unit.new(unit_name, aliases: aliases, value: value)
+  end
+
+  def case_sensitive(is_case_sensitive)
+    @case_sensitive = is_case_sensitive
+  end
+
+  def conversion
+    raise Measured::UnitError, "A base unit has not yet been set." unless @base_unit
+    check_for_duplicate_unit_names
+
+    Measured::Conversion.new(@base_unit, @units, case_sensitive: @case_sensitive)
+  end
+
+  private
+
+  def check_for_duplicate_unit_names
+    names = @base_unit.names + @units.flat_map { |unit| unit.names }
+    duplicated_names = names.group_by(&:to_s).select{ |k, v| v.size > 1 }.keys
```

Comment:
Ahhh, good catch. I'll add some test cases that exercise the building of a case-insensitive conversion.